src/types/fieldExtractor.ts

```diff
@@ -6,6 +6,7 @@ import { getTrait, NFT, NftFactory } from './nft';
 type Extractor = (nft: NFT) => string;
 type StrategyExtractor = (nftFactory: NftFactory) => Extractor;
 type Resolver = (nft: NFT, contract: NftFactory) => string;
+type IDResolver = (nft: NFT, contract: NftFactory) => string | null;
```

Comment:
Slightly odd to have a null ID but I think it's something to tackle at a later stage.